lms/djangoapps/teams/serializers.py

```diff
@@ -108,8 +112,43 @@ class Meta(object):
         read_only_fields = ("date_joined",)
 
 
-class TopicSerializer(serializers.Serializer):
-    """Serializes a topic."""
+class BaseTopicSerializer(serializers.Serializer):
+    """Serializes a topic without team_count."""
     description = serializers.CharField()
     name = serializers.CharField()
     id = serializers.CharField()  # pylint: disable=invalid-name
+
+
+class TopicSerializer(BaseTopicSerializer):
+    """
+    Adds team_count to the basic topic serializer.  Use only when
+    serializing a single topic.  When serializing many topics, use
+    `PaginatedTopicSerializer` to avoid O(N) SQL queries.
+    """
+    team_count = serializers.SerializerMethodField('get_team_count')
+
+    def get_team_count(self, topic):
+        """Get the number of teams associated with this topic"""
+        return CourseTeam.objects.filter(topic_id=topic['id']).count()
+
+
+class PaginatedTopicSerializer(PaginationSerializer):
```

Comment:
I think you are assuming the behavior of the implementation. It isn't a given that this logic works to my mind, and there could be bugs that only show up during pagination, e.g. an error when the code expects a certain topic to be there, but it isn't because it isn't in the current paginated set.